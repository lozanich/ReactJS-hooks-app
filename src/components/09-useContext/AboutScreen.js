import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const handleLogout = () => {
    setUser({});
  };
  return (
    <div>
      <h1>About Screen</h1>
      <hr />

      <pre>{JSON.stringify(user)}</pre>

      <button onClick={handleLogout} className="btn btn-warning">
        Logout
      </button>
    </div>
  );
};
