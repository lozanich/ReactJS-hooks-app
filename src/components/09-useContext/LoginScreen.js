import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const user = {
    id: 123838,
    name: "Ivan Lozano",
    email: "lozanich598@gmail.com",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button onClick={() => setUser(user)} className="btn btn-primary">
        Login
      </button>
    </div>
  );
};
