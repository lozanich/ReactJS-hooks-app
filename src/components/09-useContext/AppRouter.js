import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { Navbar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen}></Route>
            <Route exact path="/about" component={AboutScreen}></Route>
            <Route exact path="/login" component={LoginScreen}></Route>

            <Route component={HomeScreen}></Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
};
