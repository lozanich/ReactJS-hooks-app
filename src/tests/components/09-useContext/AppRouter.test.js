import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en el componente <AppRouter />", () => {
  const user = { id: 1, name: "Ivan" };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test("Debe de mostarrse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
