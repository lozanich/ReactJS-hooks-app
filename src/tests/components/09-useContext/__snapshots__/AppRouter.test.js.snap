// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRouter /> Debe de mostarrse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      UseContext
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Ivan"
}
      </pre>
    </div>
  </div>
</div>
`;
